packages/runtime-dom/src/patchProp.ts

```diff
@@ -60,7 +60,11 @@ export const patchProp: DOMRendererOptions['patchProp'] = (
   } else if (
     // #11081 force set props for possible async custom element
     (el as VueElement)._isVueCE &&
-    (/[A-Z]/.test(key) || !isString(nextValue))
+    // #12408 check if it's hyphen prop or it's async custom element
+    (camelize(key) in el ||
+      // @ts-expect-error _def is private
+      ((el as VueElement)._def.__asyncLoader &&
```

Comment:
Could you provide more info on why is this check added?

Reply:
If it's not async custom element, we should check if this key is a defined prop for below case
```
foo-bar={true}
```
if it's async, simply follow previous logic, check if it's a key containing capitals or the value is not a string

for other cases, skip. it will be patched as attribute, so that `tabindex={1}` or `data-test={true}` can work